### ISSUE TITLE
Cassandra Sink: exception java.lang.NoClassDefFoundError: Could not initialize class com.landoop.json.sql.JacksonJson$

### ISSUE THREAD

**Issue by z11373** [lensesio/stream-reactor#760]
Hi, we are running Kafka Connect Cassandra Sink connector using com.datamountaineer.kafka-connect-cassandra **v2.0.1** and got this exception below. I wonder if any of you ever seen it before.
Looks like it failed when trying to insert the data from Kafka topic, which we're using JSON format.
The KCQL is pretty straightforward, i.e. **"INSERT INTO tick_events SELECT event_id, event_timestamp, event_type, tick_price, tick_quantity, tick_symbol FROM kf-test-sink"**
I checked everything is correct, do you know what might be the culprit? Thanks!

```
c.d.s.c.c.s.CassandraJsonWriter | There was an error writing the records Boxed Error
java.util.concurrent.ExecutionException: Boxed Error
        at scala.concurrent.impl.Promise$.resolver(Promise.scala:87)
        at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:79)
        at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:284)
        at scala.concurrent.Promise.complete(Promise.scala:53)
        at scala.concurrent.Promise.complete$(Promise.scala:52)
        at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
        at scala.concurrent.Promise.failure(Promise.scala:104)
        at scala.concurrent.Promise.failure$(Promise.scala:104)
        at scala.concurrent.impl.Promise$DefaultPromise.failure(Promise.scala:187)
        at com.datamountaineer.streamreactor.connect.concurrent.ExecutorExtension$RunnableWrapper$$anon$1.run(ExecutorExtension.scala:33)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.landoop.json.sql.JacksonJson$
        at com.datamountaineer.streamreactor.connect.converters.Transform$.apply(Transform.scala:93)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2(CassandraJsonWriter.scala:187)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2$adapted(CassandraJsonWriter.scala:181)
        at scala.collection.immutable.Map$Map1.foreach(Map.scala:128)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.insert(CassandraJsonWriter.scala:181)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4(CassandraJsonWriter.scala:161)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4$adapted(CassandraJsonWriter.scala:159)
        at scala.collection.Iterator.foreach(Iterator.scala:941)
        at scala.collection.Iterator.foreach$(Iterator.scala:941)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
        at scala.collection.IterableLike.foreach(IterableLike.scala:74)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$3(CassandraJsonWriter.scala:159)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at com.datamountaineer.streamreactor.connect.concurrent.ExecutorExtension$RunnableWrapper$$anon$1.run(ExecutorExtension.scala:30)
        ... 3 common frames omitted
```

**Comment by z11373** [comment]
it's weird, now i got slightly different exception :-(

```
java.util.concurrent.ExecutionException: Boxed Error
        at scala.concurrent.impl.Promise$.resolver(Promise.scala:87)
        at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:79)
        at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:284)
        at scala.concurrent.Promise.complete(Promise.scala:53)
        at scala.concurrent.Promise.complete$(Promise.scala:52)
        at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
        at scala.concurrent.Promise.failure(Promise.scala:104)
        at scala.concurrent.Promise.failure$(Promise.scala:104)
        at scala.concurrent.impl.Promise$DefaultPromise.failure(Promise.scala:187)
        at com.datamountaineer.streamreactor.connect.concurrent.ExecutorExtension$RunnableWrapper$$anon$1.run(ExecutorExtension.scala:33)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NoSuchMethodError: 'void scala.runtime.Statics.releaseFence()'
        at com.landoop.json.sql.JacksonJson$$anon$1.<init>(JacksonJson.scala:25)
        at com.landoop.json.sql.JacksonJson$.<clinit>(JacksonJson.scala:25)
        at com.datamountaineer.streamreactor.connect.converters.Transform$.apply(Transform.scala:93)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2(CassandraJsonWriter.scala:187)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2$adapted(CassandraJsonWriter.scala:181)
        at scala.collection.immutable.Map$Map1.foreach(Map.scala:128)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.insert(CassandraJsonWriter.scala:181)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4(CassandraJsonWriter.scala:161)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4$adapted(CassandraJsonWriter.scala:159)
        at scala.collection.Iterator.foreach(Iterator.scala:941)
        at scala.collection.Iterator.foreach$(Iterator.scala:941)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
        at scala.collection.IterableLike.foreach(IterableLike.scala:74)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
        at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$3(CassandraJsonWriter.scala:159)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at com.datamountaineer.streamreactor.connect.concurrent.ExecutorExtension$RunnableWrapper$$anon$1.run(ExecutorExtension.scala:30)
        ... 3 common frames omitted
```

**Comment by z11373** [comment]
I downloaded the jar of latest version 2.1.3, and run it via connect standalone, and it got this exception:
```
[2021-04-14 14:24:03,326] ERROR There was an error writing the records Boxed Error (com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter:174)
java.util.concurrent.ExecutionException: Boxed Error
	at scala.concurrent.impl.Promise$.resolver(Promise.scala:87)
	at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:79)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:284)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.Promise.failure(Promise.scala:104)
	at scala.concurrent.Promise.failure$(Promise.scala:104)
	at scala.concurrent.impl.Promise$DefaultPromise.failure(Promise.scala:187)
	at com.datamountaineer.streamreactor.connect.concurrent.ExecutorExtension$RunnableWrapper$$anon$1.run(ExecutorExtension.scala:33)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ExceptionInInitializerError
	at com.datamountaineer.streamreactor.connect.converters.Transform$.apply(Transform.scala:93)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2(CassandraJsonWriter.scala:187)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2$adapted(CassandraJsonWriter.scala:181)
	at scala.collection.immutable.Map$Map1.foreach(Map.scala:154)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.insert(CassandraJsonWriter.scala:181)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4(CassandraJsonWriter.scala:161)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4$adapted(CassandraJsonWriter.scala:159)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$3(CassandraJsonWriter.scala:159)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.datamountaineer.streamreactor.connect.concurrent.ExecutorExtension$RunnableWrapper$$anon$1.run(ExecutorExtension.scala:30)
	... 3 more
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.9.9 requires Jackson Databind version >= 2.9.0 and < 2.10.0
	at com.fasterxml.jackson.module.scala.JacksonModule.setupModule(JacksonModule.scala:61)
	at com.fasterxml.jackson.module.scala.JacksonModule.setupModule$(JacksonModule.scala:46)
	at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:17)
	at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:808)
	at com.landoop.json.sql.JacksonJson$.<init>(JacksonJson.scala:26)
	at com.landoop.json.sql.JacksonJson$.<clinit>(JacksonJson.scala)
	... 19 more
```

**Comment by davidsloan** [comment]
Hello, version 3.0.0 has been released with a lot of major changes.  Would you like to try and let me know if this problem still persists?

**Comment by jdeex** [comment]
I have updated from 2.1.3 (was working fine) to 3.0.1 and we are now getting the same error:

` [2021-12-15 21:06:29,222] ERROR There was an error writing the records Boxed Error (com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter)
 java.util.concurrent.ExecutionException: Boxed Error
    at scala.concurrent.impl.Promise$.resolver(Promise.scala:87)
    at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:79)
    at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:284)
    at scala.concurrent.Promise.complete(Promise.scala:53)
    at scala.concurrent.Promise.complete$(Promise.scala:52)
    at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
    at scala.concurrent.Promise.failure(Promise.scala:104)
    at scala.concurrent.Promise.failure$(Promise.scala:104)
    at scala.concurrent.impl.Promise$DefaultPromise.failure(Promise.scala:187)
    at com.datamountaineer.streamreactor.common.concurrent.ExecutorExtension$RunnableWrapper$.$anonfun$submit$1(ExecutorExtension.scala:34)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.base/java.lang.Thread.run(Thread.java:829)
 Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.landoop.json.sql.JacksonJson$
    at com.datamountaineer.streamreactor.common.converters.ToJsonWithProjections$.apply(ToJsonWithProjections.scala:97)
    at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2(CassandraJsonWriter.scala:185)
    at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2$adapted(CassandraJsonWriter.scala:179)
    at scala.collection.immutable.Map$Map1.foreach(Map.scala:154)
    at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.insert(CassandraJsonWriter.scala:179)
    at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4(CassandraJsonWriter.scala:159)
    at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4$adapted(CassandraJsonWriter.scala:157)
    at scala.collection.Iterator.foreach(Iterator.scala:943)
    at scala.collection.Iterator.foreach$(Iterator.scala:943)
    at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
    at scala.collection.IterableLike.foreach(IterableLike.scala:74)
    at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
    at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
    at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$3(CassandraJsonWriter.scala:157)
    at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
    at com.datamountaineer.streamreactor.common.concurrent.ExecutorExtension$RunnableWrapper$.$anonfun$submit$1(ExecutorExtension.scala:31)
    ... 3 more`

**Comment by davidsloan** [comment]
> I have updated from 2.1.3 (was working fine) to 3.0.1 and we are now getting the same error:
> 
> ```
> [2021-12-15 21:06:29,222] ERROR There was an error writing the records Boxed Error (com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter)
>  java.util.concurrent.ExecutionException: Boxed Error
>     at scala.concurrent.impl.Promise$.resolver(Promise.scala:87)
>     at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:79)
>     at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:284)
>     at scala.concurrent.Promise.complete(Promise.scala:53)
>     at scala.concurrent.Promise.complete$(Promise.scala:52)
>     at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
>     at scala.concurrent.Promise.failure(Promise.scala:104)
>     at scala.concurrent.Promise.failure$(Promise.scala:104)
>     at scala.concurrent.impl.Promise$DefaultPromise.failure(Promise.scala:187)
>     at com.datamountaineer.streamreactor.common.concurrent.ExecutorExtension$RunnableWrapper$.$anonfun$submit$1(ExecutorExtension.scala:34)
>     at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
>     at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
>     at java.base/java.lang.Thread.run(Thread.java:829)
>  Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.landoop.json.sql.JacksonJson$
>     at com.datamountaineer.streamreactor.common.converters.ToJsonWithProjections$.apply(ToJsonWithProjections.scala:97)
>     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2(CassandraJsonWriter.scala:185)
>     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2$adapted(CassandraJsonWriter.scala:179)
>     at scala.collection.immutable.Map$Map1.foreach(Map.scala:154)
>     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.insert(CassandraJsonWriter.scala:179)
>     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4(CassandraJsonWriter.scala:159)
>     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4$adapted(CassandraJsonWriter.scala:157)
>     at scala.collection.Iterator.foreach(Iterator.scala:943)
>     at scala.collection.Iterator.foreach$(Iterator.scala:943)
>     at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
>     at scala.collection.IterableLike.foreach(IterableLike.scala:74)
>     at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
>     at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
>     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$3(CassandraJsonWriter.scala:157)
>     at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
>     at com.datamountaineer.streamreactor.common.concurrent.ExecutorExtension$RunnableWrapper$.$anonfun$submit$1(ExecutorExtension.scala:31)
>     ... 3 more
> ```

@jdeex You don't have multiple versions of the connector jar file in the plugin directory do you?

**Comment by jdeex** [comment]
> > I have updated from 2.1.3 (was working fine) to 3.0.1 and we are now getting the same error:
> > ```
> > [2021-12-15 21:06:29,222] ERROR There was an error writing the records Boxed Error (com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter)
> >  java.util.concurrent.ExecutionException: Boxed Error
> >     at scala.concurrent.impl.Promise$.resolver(Promise.scala:87)
> >     at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:79)
> >     at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:284)
> >     at scala.concurrent.Promise.complete(Promise.scala:53)
> >     at scala.concurrent.Promise.complete$(Promise.scala:52)
> >     at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
> >     at scala.concurrent.Promise.failure(Promise.scala:104)
> >     at scala.concurrent.Promise.failure$(Promise.scala:104)
> >     at scala.concurrent.impl.Promise$DefaultPromise.failure(Promise.scala:187)
> >     at com.datamountaineer.streamreactor.common.concurrent.ExecutorExtension$RunnableWrapper$.$anonfun$submit$1(ExecutorExtension.scala:34)
> >     at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
> >     at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> >     at java.base/java.lang.Thread.run(Thread.java:829)
> >  Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.landoop.json.sql.JacksonJson$
> >     at com.datamountaineer.streamreactor.common.converters.ToJsonWithProjections$.apply(ToJsonWithProjections.scala:97)
> >     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2(CassandraJsonWriter.scala:185)
> >     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2$adapted(CassandraJsonWriter.scala:179)
> >     at scala.collection.immutable.Map$Map1.foreach(Map.scala:154)
> >     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.insert(CassandraJsonWriter.scala:179)
> >     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4(CassandraJsonWriter.scala:159)
> >     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4$adapted(CassandraJsonWriter.scala:157)
> >     at scala.collection.Iterator.foreach(Iterator.scala:943)
> >     at scala.collection.Iterator.foreach$(Iterator.scala:943)
> >     at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
> >     at scala.collection.IterableLike.foreach(IterableLike.scala:74)
> >     at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
> >     at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
> >     at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$3(CassandraJsonWriter.scala:157)
> >     at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
> >     at com.datamountaineer.streamreactor.common.concurrent.ExecutorExtension$RunnableWrapper$.$anonfun$submit$1(ExecutorExtension.scala:31)
> >     ... 3 more
> > ```
> 
> @jdeex You don't have multiple versions of the connector jar file in the plugin directory do you?

No, just the 3.0.1 but we do have other connectors there: Cassandra and Elastic6

**Comment by jdeex** [comment]
Having both Cassandra and Elastic6, Elastic6 also fails with a similar exception

```
connect_1  | [2021-12-16 14:17:30,943] ERROR WorkerSinkTask{id=qos-sink-connector-0} Task threw an uncaught and unrecoverable exception. Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask)
connect_1  | org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
connect_1  |    at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:631)
connect_1  |    at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:333)
connect_1  |    at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:234)
connect_1  |    at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:203)
connect_1  |    at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:188)
connect_1  |    at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:243)
connect_1  |    at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
connect_1  |    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
connect_1  |    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
connect_1  |    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
connect_1  |    at java.base/java.lang.Thread.run(Thread.java:829)
connect_1  | Caused by: java.lang.ExceptionInInitializerError
connect_1  |    at com.datamountaineer.streamreactor.connect.elastic6.Transform$.apply(Transform.scala:94)
connect_1  |    at com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter.$anonfun$insert$6(ElasticJsonWriter.scala:112)
connect_1  |    at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:273)
connect_1  |    at scala.collection.Iterator.foreach(Iterator.scala:943)
connect_1  |    at scala.collection.Iterator.foreach$(Iterator.scala:943)
connect_1  |    at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
connect_1  |    at scala.collection.IterableLike.foreach(IterableLike.scala:74)
connect_1  |    at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
connect_1  |    at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
connect_1  |    at scala.collection.TraversableLike.map(TraversableLike.scala:273)
connect_1  |    at scala.collection.TraversableLike.map$(TraversableLike.scala:266)
connect_1  |    at scala.collection.AbstractTraversable.map(Traversable.scala:108)
connect_1  |    at com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter.$anonfun$insert$5(ElasticJsonWriter.scala:104)
connect_1  |    at scala.collection.Iterator$$anon$10.next(Iterator.scala:461)
connect_1  |    at scala.collection.Iterator.foreach(Iterator.scala:943)
connect_1  |    at scala.collection.Iterator.foreach$(Iterator.scala:943)
connect_1  |    at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
connect_1  |    at scala.collection.generic.Growable.$plus$plus$eq(Growable.scala:62)
connect_1  |    at scala.collection.generic.Growable.$plus$plus$eq$(Growable.scala:53)
connect_1  |    at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:105)
connect_1  |    at scala.collection.mutable.ArrayBuffer.$plus$plus$eq(ArrayBuffer.scala:49)
connect_1  |    at scala.collection.mutable.Builder$$anon$1.$plus$plus$eq(Builder.scala:139)
connect_1  |    at scala.collection.mutable.Builder$$anon$1.$plus$plus$eq(Builder.scala:135)
connect_1  |    at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:280)
connect_1  |    at scala.collection.mutable.ArraySeq.foreach(ArraySeq.scala:75)
connect_1  |    at scala.collection.TraversableLike.flatMap(TraversableLike.scala:280)
connect_1  |    at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:277)
connect_1  |    at scala.collection.AbstractTraversable.flatMap(Traversable.scala:108)
connect_1  |    at com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter.$anonfun$insert$1(ElasticJsonWriter.scala:98)
connect_1  |    at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:280)
connect_1  |    at scala.collection.immutable.Map$Map1.foreach(Map.scala:154)
connect_1  |    at scala.collection.TraversableLike.flatMap(TraversableLike.scala:280)
connect_1  |    at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:277)
connect_1  |    at scala.collection.AbstractTraversable.flatMap(Traversable.scala:108)
connect_1  |    at com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter.insert(ElasticJsonWriter.scala:93)
connect_1  |    at com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter.write(ElasticJsonWriter.scala:83)
connect_1  |    at com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask.$anonfun$put$2(ElasticSinkTask.scala:67)
connect_1  |    at com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask.$anonfun$put$2$adapted(ElasticSinkTask.scala:67)
connect_1  |    at scala.Option.foreach(Option.scala:407)
connect_1  |    at com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask.put(ElasticSinkTask.scala:67)
connect_1  |    at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:601)
connect_1  |    ... 10 more
connect_1  | Caused by: com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.12.3 requires Jackson Databind version >= 2.12.0 and < 2.13.0
connect_1  |    at com.fasterxml.jackson.module.scala.JacksonModule.setupModule(JacksonModule.scala:61)
connect_1  |    at com.fasterxml.jackson.module.scala.JacksonModule.setupModule$(JacksonModule.scala:46)
connect_1  |    at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:17)
connect_1  |    at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:808)
connect_1  |    at com.landoop.json.sql.JacksonJson$.<init>(JacksonJson.scala:26)
connect_1  |    at com.landoop.json.sql.JacksonJson$.<clinit>(JacksonJson.scala)
connect_1  |    ... 51 more
```

**Comment by davidsloan** [comment]
Thanks for reporting @jdeex .  I just pushed a commit designed to repair this dependency issue.

Are you able to try it from the branch 'fix/760-dependency-repair'?

I can provide you the build for testing if you get in touch on our community Slack.

**Comment by jdeex** [comment]
Sure I can test it. is this for both: Cassandra and Elastic6 or just Cassandra?

We are downloading each binary (for each connector) when building our connect container, is your fix available to download at any url? or you prefer to pass it on Slack?

https://github.com/lensesio/stream-reactor/releases/download/3.0.1/kafka-connect-cassandra-3.0.1-2.5.0-all.tar.gz
https://github.com/lensesio/stream-reactor/releases/download/3.0.1/kafka-connect-elastic6-3.0.1-2.5.0-all.tar.gz

**Comment by davidsloan** [comment]
Thank you.

No problem I can share them temporarily on Google Drive.

Cassandra: https://drive.google.com/file/d/1bpbzP0RbGql4xB_wyp5ZRa9NB-x-ukok/view?usp=sharing
Elastic6: https://drive.google.com/file/d/1AohORRdoaRRoLTpTnZ5JDLfoD_mci3dJ/view?usp=sharing

Let me know how it goes.

**Comment by jdeex** [comment]
Ok First test with Elastic6 no longer has the error:

The **_2.1.3 version_** in the following line got my attention though:
`
INFO Instantiated connector qos-sink-connector with version 2.1.3 of type class com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector`

Full log for Elasctic6:

```
kafka-connect-1  | [2021-12-17 16:39:19,189] INFO Loading plugin from: /usr/local/share/kafka/plugins/kafka-connect-elastic6-3.0.2-SNAPSHOT-2.5.0-all.jar (org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader)
kafka-connect-1  | [2021-12-17 16:39:23,401] INFO Registered loader: PluginClassLoader{pluginLocation=file:/usr/local/share/kafka/plugins/kafka-connect-elastic6-3.0.2-SNAPSHOT-2.5.0-all.jar} (org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader)
kafka-connect-1  | [2021-12-17 16:39:23,401] INFO Added plugin 'com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector' (org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader)
kafka-connect-1  | [2021-12-17 16:39:27,880] INFO Added aliases 'ElasticSinkConnector' and 'ElasticSink' to plugin 'com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector' (org.apache.kafka.connect.runtime.isolation.DelegatingClassLoader)
kafka-connect-1  | [2021-12-17 16:39:45,442] INFO Creating connector qos-sink-connector of type com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  |      connector.class = com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector
kafka-connect-1  |      connector.class = com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector
kafka-connect-1  | [2021-12-17 16:39:45,448] INFO Instantiated connector qos-sink-connector with version 2.1.3 of type class com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  | [2021-12-17 16:39:45,450] INFO Starting Elastic sink task. (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector)
kafka-connect-1  |      connector.class = com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector
kafka-connect-1  |      connector.class = com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector
kafka-connect-1  | [2021-12-17 16:39:45,873] INFO Setting task configurations for 1 workers. (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector)
kafka-connect-1  |      connector.class = com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector
kafka-connect-1  |      connector.class = com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector
kafka-connect-1  |      task.class = class com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask
kafka-connect-1  | [2021-12-17 16:39:45,942] INFO Instantiated task qos-sink-connector-0 with version 2.1.3 of type com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  |      connector.class = com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector
kafka-connect-1  |      connector.class = com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector
kafka-connect-1  |  2.1.3 (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  |        (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  |      connect.elastic.batch.size = 4000
kafka-connect-1  |      connect.elastic.cluster.name = elasticsearch
kafka-connect-1  |      connect.elastic.error.policy = RETRY
kafka-connect-1  |      connect.elastic.hosts = elasticsearch
kafka-connect-1  |      connect.elastic.kcql = INSERT INTO qos SELECT * FROM qos WITHDOCTYPE=_doc
kafka-connect-1  |      connect.elastic.max.retries = 15
kafka-connect-1  |      connect.elastic.pk.separator = -
kafka-connect-1  |      connect.elastic.port = 9200
kafka-connect-1  |      connect.elastic.protocol = https
kafka-connect-1  |      connect.elastic.retry.interval = 60000
kafka-connect-1  |      connect.elastic.tableprefix =
kafka-connect-1  |      connect.elastic.use.http.password = *******
kafka-connect-1  |      connect.elastic.use.http.username = *******
kafka-connect-1  |      connect.elastic.write.timeout = 300000
kafka-connect-1  |  (com.datamountaineer.streamreactor.connect.elastic6.config.ElasticConfig)
kafka-connect-1  | [2021-12-17 16:39:46,377] INFO Creating HTTP client on https://elasticsearch:9200 (com.sksamuel.elastic4s.http.ElasticClient$)
kafka-connect-1  | [2021-12-17 16:39:46,496] INFO Initialising Elastic Json writer (com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter)
kafka-connect-1  | [2021-12-17 16:39:56,145] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:40:06,146] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:40:16,146] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:40:26,147] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:40:36,148] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:40:46,148] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:40:56,149] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:41:06,150] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:41:16,152] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:41:26,152] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:41:30,769] WARN request [POST https://elasticsearch:9200/_bulk?refresh=true] returned 1 warnings: [299 Elasticsearch-7.10.2-747e1cc71def077253878a59143c1f785afa92b9 "[types removal] Specifying types in bulk requests is deprecated."] (org.elasticsearch.client.RestClient)
kafka-connect-1  | [2021-12-17 16:41:36,152] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)
kafka-connect-1  | [2021-12-17 16:41:46,155] INFO Flushing Elastic Sink (com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask)

```

**Comment by davidsloan** [comment]
> Ok First test with Elastic6 no longer has the error:
> 
> The **_2.1.3 version_** in the following line got my attention though: ` INFO Instantiated connector qos-sink-connector with version 2.1.3 of type class com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkConnector`
> 
Sorry about that.  That's because it's a snapshot build and not reporting the correct version number.  It's definitely the latest, secure code.

**Comment by jdeex** [comment]
Same for Cassandra:

```
kafka-connect-1  | [2021-12-17 16:54:59,170] INFO [Worker clientId=connect-1, groupId=connect-group] Connector TestSinkConnector config updated (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
kafka-connect-1  | [2021-12-17 16:54:59,218] INFO [Worker clientId=connect-1, groupId=connect-group] Joined group at generation 6 with protocol version 2 and got assignment: Assignment{error=0, leader='connect-1-04f78ece-c2f3-478d-98be-49498b61d492', leaderUrl='http://localhost:8083/', offset=11, connectorIds=[TestSinkConnector], taskIds=[TestSinkConnector-0], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
kafka-connect-1  | [2021-12-17 16:54:59,219] INFO [Worker clientId=connect-1, groupId=connect-group] Starting connector TestSinkConnector (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
kafka-connect-1  | [2021-12-17 16:54:59,220] INFO Creating connector TestSinkConnector of type com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraSinkConnector (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  |      name = TestSinkConnector
kafka-connect-1  |      name = TestSinkConnector
kafka-connect-1  | [2021-12-17 16:54:59,225] INFO Instantiated connector TestSinkConnector with version 2.1.3 of type class com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraSinkConnector (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  | [2021-12-17 16:54:59,227] INFO Finished creating connector TestSinkConnector (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  |      name = TestSinkConnector
kafka-connect-1  |      name = TestSinkConnector
kafka-connect-1  | [2021-12-17 16:54:59,517] INFO [Worker clientId=connect-1, groupId=connect-group] Tasks [TestSinkConnector-0] configs updated (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
kafka-connect-1  | [2021-12-17 16:54:59,528] INFO [Worker clientId=connect-1, groupId=connect-group] Joined group at generation 7 with protocol version 2 and got assignment: Assignment{error=0, leader='connect-1-04f78ece-c2f3-478d-98be-49498b61d492', leaderUrl='http://localhost:8083/', offset=13, connectorIds=[TestSinkConnector], taskIds=[TestSinkConnector-0], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
kafka-connect-1  | [2021-12-17 16:54:59,529] INFO [Worker clientId=connect-1, groupId=connect-group] Starting task TestSinkConnector-0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
kafka-connect-1  | [2021-12-17 16:54:59,529] INFO Creating task TestSinkConnector-0 (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  |      name = TestSinkConnector
kafka-connect-1  |      name = TestSinkConnector
kafka-connect-1  | [2021-12-17 16:54:59,534] INFO Instantiated task TestSinkConnector-0 with version 2.1.3 of type com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraSinkTask (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  | [2021-12-17 16:54:59,535] INFO Set up the key converter class org.apache.kafka.connect.storage.StringConverter for task TestSinkConnector-0 using the connector config (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  | [2021-12-17 16:54:59,535] INFO Set up the value converter class org.apache.kafka.connect.json.JsonConverter for task TestSinkConnector-0 using the connector config (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  | [2021-12-17 16:54:59,535] INFO Set up the header converter class org.apache.kafka.connect.storage.SimpleHeaderConverter for task TestSinkConnector-0 using the worker config (org.apache.kafka.connect.runtime.Worker)
kafka-connect-1  |      name = TestSinkConnector
kafka-connect-1  |      name = TestSinkConnector
kafka-connect-1  |      client.id = connector-consumer-TestSinkConnector-0
kafka-connect-1  |      group.id = connect-TestSinkConnector
```

**Comment by davidsloan** [comment]
ok, as long as everything seems to be working ok (ignore the version number) we can go forward with the fix, and get an official release out soon.

**Comment by jdeex** [comment]
Also tested the Cassandra Source and works fine now:
`
kafka-connect-1  | [2021-12-17 17:18:00,732] INFO Sending 1 records for connector TestSourceConnector (com.datamountaineer.streamreactor.connect.cassandra.source.CassandraSourceTask)`

**Comment by jdeex** [comment]
> ok, as long as everything seems to be working ok (ignore the version number) we can go forward with the fix, and get an official release out soon.

Any tentative date for the release?

**Comment by davidsloan** [comment]
> > ok, as long as everything seems to be working ok (ignore the version number) we can go forward with the fix, and get an official release out soon.
> 
> Any tentative date for the release?

New year - we have some other issues to work through with the dependencies.

**Comment by wawanopoulos** [comment]
Hi, any ETA for resolution of this issue ?

**Comment by jdeex** [comment]
@davidsloan provided a fix in a snapshot, still waiting for a full release that includes that fix

**Comment by beer1970** [comment]
> @davidsloan provided a fix in a snapshot, still waiting for a full release that includes that fix

Yes I downloaded the snapshot from https://drive.google.com/file/d/1bpbzP0RbGql4xB_wyp5ZRa9NB-x-ukok/view?usp=sharing

It fixed the error 

`java.lang.NoClassDefFoundError: Could not initialize class com.landoop.json.sql.JacksonJson`

 but it introduced another one 

`com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.12.3 requires Jackson Databind version >= 2.12.0 and < 2.13.0`

 but maybe its a configuration error as @z11373 showed the same error in the beginning of this thread.

**Comment by jdeex** [comment]
For us the snapshot works well and fixed both errors: NoClassDefFoundError and JsonMappingException

I also tested both Cassandra and Elastic and both worked well... but if in your use case there are still errors, upload logs and config for them to review

**Comment by beer1970** [comment]
I restarted the connector, seems now its more a data error, below the config. and attached the trace output

```
 config:
    connect.cassandra.contact.points: cassandra-dc1-dev-enai-service.cassandra
    connect.cassandra.kcql: INSERT INTO vlog SELECT kafka_key, kafka_value FROM dev-xxx-vlog
    connect.cassandra.key.space: xxx
    connect.cassandra.password: xxx
    connect.cassandra.username: cassandra-superuser
    connect.progress.enabled: true
    key.converter: org.apache.kafka.connect.storage.StringConverter
    key.converter.schemas.enable: false
    tasks.max: 1
    topics: dev-xxx-vlog
    value.converter: org.apache.kafka.connect.storage.StringConverter
    value.converter.schemas.enable: false

```
[log-cansandra-sink.log](https://github.com/lensesio/stream-reactor/files/8387248/log-cansandra-sink.log)

**Comment by davidsloan** [comment]
@beer1970 please can you open a new issue for your problem?  It doesn't appear to be related to the packaging issues that this ticket is about.

**Comment by vvash94** [comment]
@davidsloan Im having the same issue as @dberardo-com with MQTT sink connector. I have reached out on the community slack channel. It would be awesome if you could guide me on how to get past this. Thanks.

**Comment by davevano** [comment]
> > i am also having this problem with MQTT sink connector
> 
> @dberardo-com The changes to potentially fix this just got merged into master branch. If you would like to help test those changes please reach out to me on our [community Slack](https://launchpass.com/lensesio) and we can get you unblocked on this.

Hello, same issue for me with MQTT Sink Coonector. This is very annoying to be limited to the 2.1.3 version.
Is there a fix planned soon?
Thanks

**Comment by davidsloan** [comment]
Have you tried building from the master branch? As I understand this is fixed.

**Comment by mparikhaiven** [comment]
I am facing the same issue
`org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:611)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:333)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:234)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:203)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:188)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:243)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.kafka.connect.errors.ConnectException: java.util.concurrent.ExecutionException: Boxed Error
	at com.datamountaineer.streamreactor.common.errors.ThrowErrorPolicy.handle(ErrorPolicy.scala:59)
	at com.datamountaineer.streamreactor.common.errors.ErrorHandler.handleError(ErrorHandler.scala:85)
	at com.datamountaineer.streamreactor.common.errors.ErrorHandler.handleTry(ErrorHandler.scala:66)
	at com.datamountaineer.streamreactor.common.errors.ErrorHandler.handleTry$(ErrorHandler.scala:48)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.handleTry(CassandraJsonWriter.scala:44)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.write(CassandraJsonWriter.scala:173)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.write(CassandraJsonWriter.scala:137)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraSinkTask.$anonfun$put$2(CassandraSinkTask.scala:73)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraSinkTask.$anonfun$put$2$adapted(CassandraSinkTask.scala:73)
	at scala.Option.foreach(Option.scala:407)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraSinkTask.put(CassandraSinkTask.scala:73)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:581)
	... 10 more
Caused by: java.util.concurrent.ExecutionException: Boxed Error
	at scala.concurrent.impl.Promise$.resolver(Promise.scala:87)
	at scala.concurrent.impl.Promise$.scala$concurrent$impl$Promise$$resolveTry(Promise.scala:79)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:284)
	at scala.concurrent.Promise.complete(Promise.scala:53)
	at scala.concurrent.Promise.complete$(Promise.scala:52)
	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
	at scala.concurrent.Promise.failure(Promise.scala:104)
	at scala.concurrent.Promise.failure$(Promise.scala:104)
	at scala.concurrent.impl.Promise$DefaultPromise.failure(Promise.scala:187)
	at com.datamountaineer.streamreactor.common.concurrent.ExecutorExtension$RunnableWrapper$.$anonfun$submit$1(ExecutorExtension.scala:34)
	... 3 more
Caused by: java.lang.ExceptionInInitializerError
	at com.datamountaineer.streamreactor.common.converters.ToJsonWithProjections$.apply(ToJsonWithProjections.scala:97)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2(CassandraJsonWriter.scala:185)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$insert$2$adapted(CassandraJsonWriter.scala:179)
	at scala.collection.immutable.Map$Map1.foreach(Map.scala:154)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.insert(CassandraJsonWriter.scala:179)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4(CassandraJsonWriter.scala:159)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$4$adapted(CassandraJsonWriter.scala:157)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at com.datamountaineer.streamreactor.connect.cassandra.sink.CassandraJsonWriter.$anonfun$write$3(CassandraJsonWriter.scala:157)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at com.datamountaineer.streamreactor.common.concurrent.ExecutorExtension$RunnableWrapper$.$anonfun$submit$1(ExecutorExtension.scala:31)
	... 3 more
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.12.6 requires Jackson Databind version >= 2.12.0 and < 2.13.0
	at com.fasterxml.jackson.module.scala.JacksonModule.setupModule(JacksonModule.scala:61)
	at com.fasterxml.jackson.module.scala.JacksonModule.setupModule$(JacksonModule.scala:46)
	at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:17)
	at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:808)
	at com.landoop.json.sql.JacksonJson$.<init>(JacksonJson.scala:26)
	at com.landoop.json.sql.JacksonJson$.<clinit>(JacksonJson.scala)
	... 19 more`

**Comment by davidsloan** [comment]
@mparikhaiven What version of the project are you using?  Which Kafka are you running against?  Have you tried building from the master branch?

**Comment by mparikhaiven** [comment]
> @mparikhaiven What version of the project are you using? Which Kafka are you running against? Have you tried building from the master branch?
@davidsloan - below is info, hope this may help.

Sink connector : v3.0.1
Kafka version : 3.1

Haven't tried on master branch as using a managed connector offering

**Comment by davidsloan** [comment]
> > @mparikhaiven What version of the project are you using? Which Kafka are you running against? Have you tried building from the master branch?
> > @davidsloan - below is info, hope this may help.
> 
> Sink connector : v3.0.1 Kafka version : 3.1
> 
> Haven't tried on master branch as using a managed connector offering

If you cannot build the connector yourself, unfortunately, you have no alternative but to wait for the next release.

**Comment by dberardo-com** [comment]
is there any forseable release date / roadmap? how to get informed about  or promote this?

**Comment by mparikhaiven** [comment]
> > > @mparikhaiven What version of the project are you using? Which Kafka are you running against? Have you tried building from the master branch?
> > > @davidsloan - below is info, hope this may help.
> > 
> > 
> > Sink connector : v3.0.1 Kafka version : 3.1
> > Haven't tried on master branch as using a managed connector offering
> 
> If you cannot build the connector yourself, unfortunately, you have no alternative but to wait for the next release.

@davidsloan - It would be great to know ETA of a release that encompasses this fix.

**Comment by kwlockwo** [comment]
I'm also seeing the same error on the InfluxDB connector (version `3.0.1`, Kafka `3.2.0`):
```
org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:611)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:333)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:234)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:203)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:188)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:243)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.landoop.json.sql.JacksonJson$
	at com.datamountaineer.streamreactor.connect.influx.converters.SinkRecordParser$.$anonfun$build$2(SinkRecordParser.scala:59)
	at scala.util.Try$.apply(Try.scala:213)
	at com.datamountaineer.streamreactor.connect.influx.converters.SinkRecordParser$.build(SinkRecordParser.scala:59)
	at com.datamountaineer.streamreactor.connect.influx.writers.InfluxBatchPointsBuilder.$anonfun$build$1(InfluxBatchPointsBuilder.scala:90)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:273)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at scala.collection.TraversableLike.map(TraversableLike.scala:273)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:266)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at com.datamountaineer.streamreactor.connect.influx.writers.InfluxBatchPointsBuilder.build(InfluxBatchPointsBuilder.scala:88)
	at com.datamountaineer.streamreactor.connect.influx.writers.InfluxDbWriter.write(InfluxDbWriter.scala:45)
	at com.datamountaineer.streamreactor.connect.influx.InfluxSinkTask.$anonfun$put$2(InfluxSinkTask.scala:77)
	at com.datamountaineer.streamreactor.connect.influx.InfluxSinkTask.$anonfun$put$2$adapted(InfluxSinkTask.scala:77)
	at scala.Option.foreach(Option.scala:407)
	at com.datamountaineer.streamreactor.connect.influx.InfluxSinkTask.put(InfluxSinkTask.scala:77)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:581)
	... 10 more
```

**Comment by aromanos** [comment]
@davidsloan is there any update on when a new release (4.0.0?) including this fix is going to be issued and new binaries provided for download? Thanks!

**Comment by davidsloan** [comment]
> @davidsloan is there any update on when a new release (4.0.0?) including this fix is going to be issued and new binaries provided for download? Thanks!

@aromanos please see here for the new release:
https://github.com/lensesio/stream-reactor/releases/tag/4.0.0

**Comment by jdeex** [comment]
Hello @davidsloan, we have tested the latest release 4.0.0 for Kafka 3.1 and we are facing the same old error for Elastic6 and Elastic7 connectors. 

See logs below:

[kafka-connect-elastic6-kafka-3-1-4.0.0.zip](https://github.com/lensesio/stream-reactor/releases/download/4.0.0/kafka-connect-elastic6-kafka-3-1-4.0.0.zip) - Scala module 2.13.3 requires Jackson Databind version >= 2.13.0 and < 2.14.0
[kafka-connect-elastic7-kafka-3-1-4.0.0.zip](https://github.com/lensesio/stream-reactor/releases/download/4.0.0/kafka-connect-elastic7-kafka-3-1-4.0.0.zip) - Scala module 2.13.3 requires Jackson Databind version >= 2.13.0 and < 2.14.0

`kafka-connect-1  | java.lang.ExceptionInInitializerError
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic7.Transform$.apply(Transform.scala:97)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic7.ElasticJsonWriter.$anonfun$insert$5(ElasticJsonWriter.scala:126)
kafka-connect-1  |      at scala.collection.immutable.Vector1.map(Vector.scala:1872)
kafka-connect-1  |      at scala.collection.immutable.Vector1.map(Vector.scala:375)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic7.ElasticJsonWriter.$anonfun$insert$4(ElasticJsonWriter.scala:119)
kafka-connect-1  |      at scala.collection.Iterator$$anon$9.next(Iterator.scala:575)
kafka-connect-1  |      at scala.collection.mutable.Growable.addAll(Growable.scala:62)
kafka-connect-1  |      at scala.collection.mutable.Growable.addAll$(Growable.scala:57)
kafka-connect-1  |      at scala.collection.mutable.ArrayBuffer.addAll(ArrayBuffer.scala:142)
kafka-connect-1  |      at scala.collection.mutable.ArrayBuffer.addAll(ArrayBuffer.scala:42)
kafka-connect-1  |      at scala.collection.mutable.GrowableBuilder.addAll(GrowableBuilder.scala:34)
kafka-connect-1  |      at scala.collection.mutable.GrowableBuilder.addAll(GrowableBuilder.scala:25)
kafka-connect-1  |      at scala.collection.mutable.Builder$$anon$1.addAll(Builder.scala:81)
kafka-connect-1  |      at scala.collection.mutable.Builder$$anon$1.addAll(Builder.scala:78)
kafka-connect-1  |      at scala.collection.StrictOptimizedIterableOps.flatMap(StrictOptimizedIterableOps.scala:117)
kafka-connect-1  |      at scala.collection.StrictOptimizedIterableOps.flatMap$(StrictOptimizedIterableOps.scala:104)
kafka-connect-1  |      at scala.collection.immutable.ArraySeq.flatMap(ArraySeq.scala:35)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic7.ElasticJsonWriter.$anonfun$insert$1(ElasticJsonWriter.scala:114)
kafka-connect-1  |      at scala.collection.StrictOptimizedIterableOps.flatMap(StrictOptimizedIterableOps.scala:117)
kafka-connect-1  |      at scala.collection.StrictOptimizedIterableOps.flatMap$(StrictOptimizedIterableOps.scala:104)
kafka-connect-1  |      at scala.collection.immutable.HashMap.flatMap(HashMap.scala:39)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic7.ElasticJsonWriter.insert(ElasticJsonWriter.scala:104)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic7.ElasticJsonWriter.write(ElasticJsonWriter.scala:95)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic7.ElasticSinkTask.$anonfun$put$2(ElasticSinkTask.scala:71)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic7.ElasticSinkTask.$anonfun$put$2$adapted(ElasticSinkTask.scala:71)
kafka-connect-1  |      at scala.Option.foreach(Option.scala:437)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic7.ElasticSinkTask.put(ElasticSinkTask.scala:71)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:581)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:333)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:234)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:203)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:188)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:243)
kafka-connect-1  |      at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
kafka-connect-1  |      at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
kafka-connect-1  |      at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
kafka-connect-1  |      at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
kafka-connect-1  |      at java.base/java.lang.Thread.run(Thread.java:829)
kafka-connect-1  | Caused by: com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.13.3 requires Jackson Databind version >= 2.13.0 and < 2.14.0
kafka-connect-1  |      at com.fasterxml.jackson.module.scala.JacksonModule.setupModule(JacksonModule.scala:61)
kafka-connect-1  |      at com.fasterxml.jackson.module.scala.JacksonModule.setupModule$(JacksonModule.scala:46)
kafka-connect-1  |      at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:17)
kafka-connect-1  |      at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:835)
kafka-connect-1  |      at com.fasterxml.jackson.databind.cfg.MapperBuilder.addModule(MapperBuilder.java:243)
kafka-connect-1  |      at com.landoop.json.sql.JacksonJson$.<clinit>(JacksonJson.scala:26)
kafka-connect-1  |      ... 38 more

#####  ELASTIC  6 CONNECTOR

kafka-connect-1  | java.lang.ExceptionInInitializerError
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic6.Transform$.apply(Transform.scala:97)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter.$anonfun$insert$6(ElasticJsonWriter.scala:122)
kafka-connect-1  |      at scala.collection.immutable.Vector1.map(Vector.scala:1872)
kafka-connect-1  |      at scala.collection.immutable.Vector1.map(Vector.scala:375)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter.$anonfun$insert$5(ElasticJsonWriter.scala:115)
kafka-connect-1  |      at scala.collection.Iterator$$anon$9.next(Iterator.scala:575)
kafka-connect-1  |      at scala.collection.mutable.Growable.addAll(Growable.scala:62)
kafka-connect-1  |      at scala.collection.mutable.Growable.addAll$(Growable.scala:57)
kafka-connect-1  |      at scala.collection.mutable.ArrayBuffer.addAll(ArrayBuffer.scala:142)
kafka-connect-1  |      at scala.collection.mutable.ArrayBuffer.addAll(ArrayBuffer.scala:42)
kafka-connect-1  |      at scala.collection.mutable.GrowableBuilder.addAll(GrowableBuilder.scala:34)
kafka-connect-1  |      at scala.collection.mutable.GrowableBuilder.addAll(GrowableBuilder.scala:25)
kafka-connect-1  |      at scala.collection.mutable.Builder$$anon$1.addAll(Builder.scala:81)
kafka-connect-1  |      at scala.collection.mutable.Builder$$anon$1.addAll(Builder.scala:78)
kafka-connect-1  |      at scala.collection.StrictOptimizedIterableOps.flatMap(StrictOptimizedIterableOps.scala:117)
kafka-connect-1  |      at scala.collection.StrictOptimizedIterableOps.flatMap$(StrictOptimizedIterableOps.scala:104)
kafka-connect-1  |      at scala.collection.immutable.ArraySeq.flatMap(ArraySeq.scala:35)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter.$anonfun$insert$1(ElasticJsonWriter.scala:109)
kafka-connect-1  |      at scala.collection.StrictOptimizedIterableOps.flatMap(StrictOptimizedIterableOps.scala:117)
kafka-connect-1  |      at scala.collection.StrictOptimizedIterableOps.flatMap$(StrictOptimizedIterableOps.scala:104)
kafka-connect-1  |      at scala.collection.immutable.HashMap.flatMap(HashMap.scala:39)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter.insert(ElasticJsonWriter.scala:99)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic6.ElasticJsonWriter.write(ElasticJsonWriter.scala:90)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask.$anonfun$put$2(ElasticSinkTask.scala:71)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask.$anonfun$put$2$adapted(ElasticSinkTask.scala:71)
kafka-connect-1  |      at scala.Option.foreach(Option.scala:437)
kafka-connect-1  |      at com.datamountaineer.streamreactor.connect.elastic6.ElasticSinkTask.put(ElasticSinkTask.scala:71)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:581)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:333)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:234)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:203)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:188)
kafka-connect-1  |      at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:243)
kafka-connect-1  |      at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
kafka-connect-1  |      at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
kafka-connect-1  |      at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
kafka-connect-1  |      at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
kafka-connect-1  |      at java.base/java.lang.Thread.run(Thread.java:829)
kafka-connect-1  | Caused by: com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.13.3 requires Jackson Databind version >= 2.13.0 and < 2.14.0
kafka-connect-1  |      at com.fasterxml.jackson.module.scala.JacksonModule.setupModule(JacksonModule.scala:61)
kafka-connect-1  |      at com.fasterxml.jackson.module.scala.JacksonModule.setupModule$(JacksonModule.scala:46)
kafka-connect-1  |      at com.fasterxml.jackson.module.scala.DefaultScalaModule.setupModule(DefaultScalaModule.scala:17)
kafka-connect-1  |      at com.fasterxml.jackson.databind.ObjectMapper.registerModule(ObjectMapper.java:835)
kafka-connect-1  |      at com.fasterxml.jackson.databind.cfg.MapperBuilder.addModule(MapperBuilder.java:243)
kafka-connect-1  |      at com.landoop.json.sql.JacksonJson$.<clinit>(JacksonJson.scala:26)`

**Comment by jdeex** [comment]
Cassandra connector works fine, though :)

[kafka-connect-cassandra-kafka-3-1-4.0.0.zip](https://github.com/lensesio/stream-reactor/releases/download/4.0.0/kafka-connect-cassandra-kafka-3-1-4.0.0.zip) OK

`kafka-connect-1  | [2022-10-06 18:30:38,603] INFO Sending 1 records for connector CSourceConnector (com.datamountaineer.streamreactor.connect.cassandra.source.CassandraSourceTask)`

**Comment by davidsloan** [comment]
I believe we have the solution for the issue now.  Let me know if anyone would like to help test the solution.

**Comment by jdeex** [comment]
> I believe we have the solution for the issue now. Let me know if anyone would like to help test the solution.

Tested with Kafka 3.3.1 error persist in the sink, the source works fine. I can test your solution.

BTW, do you suppport Cassandra 4?

**Comment by davidsloan** [comment]
Hi @jdeex .

Thanks for testing our test version and letting us know that the class loading issues have been resolved.

This has now been officially released as 4.1.0.

I've also raised an internal task to make the "connect.progress.enabled" documentation tweak.

I'll close this, but please let me know if you experience any further problems.
